packages/remix-react/links.ts

```diff
@@ -282,60 +285,31 @@ export async function getStylesheetPrefetchLinks(
 // This is ridiculously identical to transition.ts `filterMatchesToLoad`
 export function getNewMatchesForLinks(
   page: string,
-  nextMatches: RouteMatch<ClientRoute>[],
-  currentMatches: RouteMatch<ClientRoute>[],
+  nextMatches: ClientMatch[],
+  currentMatches: ClientMatch[],
   location: Location,
   mode: "data" | "assets"
-): RouteMatch<ClientRoute>[] {
+): ClientMatch[] {
   let path = parsePathPatch(page);
 
-  let isNew = (match: RouteMatch<ClientRoute>, index: number) => {
-    if (!currentMatches[index]) return true;
-    return match.route.id !== currentMatches[index].route.id;
-  };
-
-  let matchPathChanged = (match: RouteMatch<ClientRoute>, index: number) => {
-    return (
-      // param change, /users/123 -> /users/456
-      currentMatches[index].pathname !== match.pathname ||
-      // splat param changed, which is not present in match.path
-      // e.g. /files/images/avatar.jpg -> files/finances.xls
-      (currentMatches[index].route.path?.endsWith("*") &&
-        currentMatches[index].params["*"] !== match.params["*"])
-    );
-  };
```

Comment:
These are private functions inside `routeMatching.ts` now, composed and exposed in a single `shouldReloadForMatch(currentMatches, match, index)` method